src/script.js

```diff
@@ -0,0 +1,388 @@
+    img: "images/josh.svg",
+    hair: "blonde",
+    eyes: "green",
+    accessories: [],
+    other: []
+  },
+  {
+    name: "Jude",
+    img: "images/jude.svg",
+    hair: "black",
+    eyes: "green",
+    accessories: [],
+    other: []
+  },
+  {
+    name: "Julie",
+    img: "images/julie.svg",
+    hair: "black",
+    eyes: "brown",
+    accessories: ["glasses", "hats"],
+    other: []
+  },
+]
+
+
+let secretCharacter;
+let currentQuestion;
+let charactersInPlay;
+let timerInterval;
+
+// Draw the game board
+const generateBoard = () => {
+  gameBoard.innerHTML = ``
+  charactersInPlay.forEach((character) => {
+    board.innerHTML += `
+      <div class="card">
+        <p class="name">${character.name}</p>
+        <img src=${character.img} alt=${character.name}>
+        <div class="guess">
+          <button class="transparent-button" onclick="guess('${character.name}')">Guess <span class="name-block">${character.name}?</span></button>
+        </div>
+      </div>
+    `
+  })
+}
+
+// Randomly select a person from the characters array 
+const setSecretCharacter = () => {
+  secretCharacter = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
+}
+
+const startGame = () => {
+  charactersInPlay = characters;
+  generateBoard();
+  setSecretCharacter();
+  guessCount = 0
+  totalGuesses.innerText = "Number of guesses: 0"
+  startTimer()
+}
+
+// setting the currentQuestion object when you select something in the dropdown
+const selectQuestion = () => {
+  questionsOptions.forEach(question => {
+    question.addEventListener("change", () => {
+      const category = question.options[0].value;
+      const value = question.options[question.selectedIndex].value;
+
+      currentQuestion = {
+        category,
+        value,
+      }
+
+      filterCharacters(currentQuestion); 
+      question.selectedIndex = 0;
+    })
+  })
+}
+
+// It will filter the characters array and redraw the game board.
+const filterCharacters = (currentQuestion) => {
+  const secretCharacterValues = secretCharacter.hair === currentQuestion.value || secretCharacter.eyes === currentQuestion.value || secretCharacter.accessories.includes(currentQuestion.value) || secretCharacter.other.includes(currentQuestion.value);
+  const isArray = currentQuestion.category === 'accessories' || currentQuestion.category === 'other';
+
+  if (isArray) {
+    charactersInPlay = charactersInPlay.filter((character) => character[currentQuestion.category].includes(currentQuestion.value) === secretCharacterValues);
+  } else {
+    charactersInPlay = charactersInPlay.filter((character) => (character[currentQuestion.category] === currentQuestion.value) === secretCharacterValues);
+  }
+
+  alertMessage(secretCharacterValues, currentQuestion.value, currentQuestion.category);
+  generateBoard();
+  guessCount++
+  totalGuesses.innerHTML = `Number of guesses: ${guessCount}`
+  messageWrapper.style.display = "block";
+}
+
+const alertMessage = (keepCharacter) => {
```

Comment:
on a newly added function like this, it would be nice with explaining comments on how you have chosen to restructure and what this function does